### PR TITLE
Remove PSR extension

### LINKED ISSUE #201: Remove or update php-psr pecl package
Hi

Is it possible to remove the `psr` pecl extension to avoid creating issues like: https://github.com/symfony/symfony/issues/40870 ?

Or just update the images so they have the latest version of the pecl extension `psr`.

### PR DESCRIPTION
**Pull request by jderusse**
Fixes #201

The PSR extension breaks projects using different versions of PSR running in devilbox base image:

The problem **can not be fixed** and is inherited from the design of the extension. Thus I suggest **removing this extension** from the image.

There is a single version of the extension for ALL PSR as a result people can not install `psr/log=1` alongside `psr/container=2`
The extension takes precedence over PHP vendors, as a result when people install a `psr/container=2`, the API from version `1` is loaded, as result, people waste hours of debugging and report bugs to OSS maintainer

### CODE REVIEW

**Comment by jbboehr** [comment]
@jderusse The PSR extension is a required dependency of the phalcon framework. I am not affiliated with devilbox, but I would suggest that instead of complete removal, neither psr nor phalcon is enabled by default, except by user request.

**Comment by jderusse** [comment]
> @jderusse The PSR extension is a required dependency of the phalcon framework. I am not affiliated with devilbox, but I would suggest that instead of complete removal, neither psr nor phalcon is enabled by default, except by user request.

Thanks for the hint and your help on this ❤️ 

If phalcon requires psr, what happen if somebody want using phalcon with psr/container:2? is it allowed, will it work?

**Comment by cytopia** [comment]
You can already disable this extension via an environment variable:
```
docker run --rm -e DISABLE_MODULES=psr -e DEBUG_ENTRYPOINT=2 devilbox/php-fpm:7.2-prod
```
Log extract then shows the following (note also the warning about phalcon)
```
[INFO] Debug level: 2
...
[INFO] $ENABLE_MODULES not set. Not enabling any PHP modules.
[INFO] Disabling the following PHP modules: psr
root $ rm /usr/local/etc/php/conf.d/docker-php-ext-psr.ini
[19-Apr-2021 22:51:12 UTC] PHP Warning:  Cannot load module 'phalcon' because required module 'psr' is not loaded in Unknown on line 0
[19-Apr-2021 22:51:12] NOTICE: fpm is running, pid 154
[19-Apr-2021 22:51:12] NOTICE: ready to handle connections
2021-04-19 22:51:13,870 INFO success: php-fpm entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
```

**Comment by jbboehr** [comment]
@jderusse I'm not a phalcon maintainer, so I can't speak for them, but AFAIK it would require changes to the phalcon extension to support psr/container v2, and of course neither phalcon nor psr extension versions can be managed by composer.

**Comment by jderusse** [comment]
PR updated to not-enabling the extension by default.

@cytopia I don't use this image, and I understand that disabling the extension will fix the issue. This PR is about helping people that use this image and will have a hard time troubleshooting the errors. (because the error is not easy to understand and debug) 

Bumping the extension to version 2 should solve the issue for people that require `psr/container:2` but it will generate the same issue for people that run a library which is not compatible with that version (ie laravel https://github.com/laravel/framework/blob/277c2fbd0cebd2cb194807654d870f4040e288c0/src/Illuminate/Container/Container.php#L184)

**Comment by cyphos1973** [comment]
Hello
when do you think about updating the docker image? I need it fairly quickly. Thank you

**Review by Nyholm — APPROVED** [review]
I think it is a great idea to make the psr and phalcon extension opt-in. 

Thank you @jderusse and @jbboehr

**Comment by cytopia** [comment]
@jderusse Unfortunately this PR will not actually do the change.

Everything inside `Dockerfiles/` will be auto-created by Ansible. This also holds true for the CI pipeline. So your changes will just be ignored and rebuild.

The definitions for how to create the Dockerfiles are within the `build/` directory. I currently have no procedure in place to explicittly disable an extension with the yaml template engine, so this is something I will have to do first. Apart from this, also currently most builds are just failing and will need to be fixed prior.

**TL;DR**

* Need to fix builds in general
* Add parser for default-disabled-but-installed modules

I will spent some time over the weekend on this and should have it ready by Monday.

Also worth noting, this PR might get closed then, but I'll keep it open for the time being, as we do have quite some discussion in here

**Comment by cyphos1973** [comment]
Hi
Excuse me.
you could find solve this problem?
Thank you for your time

**Comment by cytopia** [comment]
Will be addressed here: https://github.com/devilbox/docker-php-fpm/pull/204

**Comment by jderusse** [comment]
closing in favor of #204
